admin/listOrdersJson/endpoint.js

```diff
@@ -1,9 +1,14 @@
-module.exports = (purchases) => (req, res) => {
+module.exports = (purchases, maerkelex) => (req, res) => {
     purchases.list((error, orders) => {
         if(error) {
             console.error("Failed to list purchases", error);
             return res.status(500).send({ error: "failed to list orders" });
         }
+        maerkelex.getData((error, maerkeData) => {
+            if(error) {
+                console.error("Failed to list purchase", error);
+                res.fail(500, "Failed to get data from maerkelex.dk");
```

Comment:
This line needs a `return` so code execution does not continue after here.

Also note that this endpoint in general sends back JSON, so using `res.fail` is inconsistent. You should send back an obejct with a field `error` just like the case above.

Having an endpoint randomly, depending on how it fails, send back content of different types is extremely confusing and so bad practice. Endpoints should be as consistent and predictable as possible.